src/client/mod.rs

```diff
@@ -25,11 +25,13 @@
 //! [`Client`]: ./struct.Client.html
 //! [`hyper::Client`]: ../../hyper/struct.Client.html
 
+mod background;
 mod connect;
 mod connection;
 mod future;
 
-pub use self::connect::{Connect, ConnectError};
+pub use self::background::Background;
```

Comment:
I don't believe the type needs to be re-exported. It can be "sealed".

Reply:
With the public `ConnectExecutor`, the only reason to have `Background` public is for completeness of documentation.

Reply:
The downside is it is now part of the public API, meaning changes to the type == breaking change.

Reply:
It was not published in a crates.io release, so it's not too late for #49 to fix this :smile: